frontend/packages/kubevirt-plugin/src/k8s/patches/vm/vm-cdrom-patches.ts

```diff
@@ -0,0 +1,149 @@
+import { last, flatten, includes } from 'lodash';
+import {
+  BOOT_ORDER_SECOND,
+  BOOT_ORDER_FIRST,
+  PVC_ACCESSMODE_RWM,
+  getBootableDevicesInOrder,
+} from 'kubevirt-web-ui-components';
+import { getName } from '@console/shared';
+import { Volume, K8sResourceKind, CD } from '@console/internal/module/k8s';
+import {
+  getVolumes,
+  getDataVolumeTemplates,
+  getDisks,
+  getVolumeDataVolumeName,
+} from '../../../selectors/vm';
+
+const getNextAvailableBootOrderIndex = (vm) => {
+  const sortedBootableDevices = getBootableDevicesInOrder(vm);
+  const numBootableDevices = sortedBootableDevices.length;
+  const lastBootableDevice: any = last(sortedBootableDevices);
+
+  // assigned indexes start at two as the first index is assigned directly by the user
+  return numBootableDevices > 0 ? lastBootableDevice.value.bootOrder + 1 : BOOT_ORDER_SECOND;
+};
+
+const assignBootOrderIndex = (vm, currDevBootOrder = -1) => {
+  let bootOrder = currDevBootOrder;
+  if (currDevBootOrder !== BOOT_ORDER_FIRST) {
+    bootOrder = getNextAvailableBootOrderIndex(vm);
+  }
+  return bootOrder;
+};
+export const getCDsPatch = (vm, cds, pvcs) => {
+  let newBootOrder = assignBootOrderIndex(vm);
+  let CDS = getDisks(vm);
+  let VOLS = getVolumes(vm);
+  let DATATEMPLATES = getDataVolumeTemplates(vm);
+
+  const patches = cds
+    .filter((cd) => cd.changed)
+    .map(({ name, pvc, type, bootOrder, bus, container, url, storageClass }) => {
+      const pvcObject = pvcs.find((vol) => getName(vol) === pvc) || pvcs[0];
+      const existingCD = !!bootOrder;
+
+      const disk: CD = {
+        name,
+        bootOrder: existingCD ? bootOrder : newBootOrder,
+        cdrom: { bus: bus || 'virtio' },
+      };
+      let volume: Volume = { name };
+      let dataVolumeTemplate: K8sResourceKind = null;
+
+      // Patches
+      if (type === 'pvc') {
+        volume = {
+          persistentVolumeClaim: {
+            claimName: getName(pvcObject),
+          },
+          name,
+        };
+      }
+
+      if (type === 'url') {
+        dataVolumeTemplate = {
+          metadata: {
+            name: `${getName(vm)}-${name}`,
+          },
+          spec: {
+            pvc: {
+              accessModes: [PVC_ACCESSMODE_RWM],
```

Comment:
I believe `kubevirt-storage-class-defaults` ConfigMap should be used here. Including setting of the `volumeMode`. 
Please consider reuse of https://github.com/kubevirt/web-ui-components/blob/master/src/k8s/objects/pvc/pvc.js .

Reply:
added `volumeMode`

Reply:
Both `accessModes` and `volumeMode` should be taken from the `kubevirt-storage-class-defaults` ConfigMap